src/nemo-bookmark-list.c

```diff
@@ -238,16 +238,19 @@ clear_bookmarks (NemoBookmarkList *bookmarks, GList *list)
 static void
 do_finalize (GObject *object)
 {
-	if (NEMO_BOOKMARK_LIST (object)->monitor != NULL) {
-		g_file_monitor_cancel (NEMO_BOOKMARK_LIST (object)->monitor);
-		NEMO_BOOKMARK_LIST (object)->monitor = NULL;
+	NemoBookmarkList *self = NEMO_BOOKMARK_LIST (object);
+
+	if (self->monitor != NULL)
+	{
+		g_file_monitor_cancel (self->monitor);
+		g_clear_object (&self->monitor);
 	}
 
-	g_queue_free (NEMO_BOOKMARK_LIST (object)->pending_ops);
+	g_queue_free (self->pending_ops);
 
-	clear_bookmarks (NEMO_BOOKMARK_LIST (object), NEMO_BOOKMARK_LIST (object)->list);
+	clear_bookmarks (self, self->list);
 
-    g_object_unref (NEMO_BOOKMARK_LIST (object)->volume_monitor);
+        g_object_unref (&self->volume_monitor);
```

Comment:
ditch the & operator - that's only required for g_clear_object() so it can set the pointer to NULL for you.